AWSS3Tests/AWSS3TransferUtilityTests.swift

```diff
@@ -2801,6 +2801,19 @@ class AWSS3TransferUtilityTests: XCTestCase {
         }
     }
 
+    // Test only verifies this code compiles with Swift.
+    // https://github.com/aws-amplify/aws-sdk-ios/issues/3725
+    func testEnumerateToAssignBlocks() {
```

Comment:
Can we add one integration test with this under `AWSS3TransferUtilityTests.swift`?

Reply:
Moved into unit tests and added 3 tests to check all of the behavior.